src/NServiceBus.Core/Pipeline/Incoming/DeserializeLogicalMessagesConnector.cs

```diff
@@ -36,7 +36,8 @@ public override async Task Invoke(IIncomingPhysicalMessageContext context, Func<
 
         static bool IsControlMessage(IncomingMessage incomingMessage)
         {
-            return incomingMessage.Headers.ContainsKey(Headers.ControlMessageHeader) && incomingMessage.Headers[Headers.ControlMessageHeader] == bool.TrueString;
+            incomingMessage.Headers.TryGetValue(Headers.ControlMessageHeader, out var value);
+            return string.Equals(value, bool.TrueString, StringComparison.OrdinalIgnoreCase);
```

Comment:
strictly speaking the `StringComparison.OrdinalIgnoreCase` is not necessary because the previous code already assumed that the code is always setting and reading the bool constants TrueStringn or FalseString

Reply:
@danielmarbach I know, it still felt for a change to the better as it doesn't hurt to be more forgiving in this case (e.g. some 3rd party integration).

Reply:
I'm not objecting to that thinking. I remember I mentioned this a few years ago and the line of thinking was that ControlMessage Header is internal to NSB anyway and will never be used as an integration message and that's why we didn't change it at that time